arm-cognitiveservices/2017-04-18/swagger/cognitiveservices.json

```diff
@@ -0,0 +1,1071 @@
+          "description": "The operation supported by Advisor."
+        }
+      }
+    },
+    "OperationDisplayInfo": {
+      "description": "The operation supported by Advisor.",
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "The description of the operation.",
+          "type": "string"
+        },
+        "operation": {
+          "description": "The action that users can perform, based on their permission level.",
+          "type": "string"
+        },
+        "provider": {
+          "description": "Service provider: Microsoft Cognitive Services.",
+          "type": "string"
+        },
+        "resource": {
+          "description": "Resource on which the operation is performed.",
+          "type": "string"
+        }
+      }
+    },
+    "CheckSkuAvailabilityParameter": {
+      "description": "Check SKU availability parameter.",
+      "properties": {
+        "sku": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Sku"
+          },
+          "description": "Check SKU parameters"
+        },
+        "kind": {
+          "$ref": "#/definitions/CognitiveServicesAccountKind"
+        },
+        "type": {
+          "type": "string",
+          "description": "resource type."
+        }
+      }
+    },
+    "CheckSkuAvailabilityResultList": {
+      "description": "Check SKU availability result list.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CheckSkuAvailabilityResult"
+          }
+        }
+      }
+    },
+    "CheckSkuAvailabilityResult": {
+      "description": "Check SKU availability result.",
+      "properties": {
+        "skuAvailable": {
+          "type": "string",
+          "description": "Indicates the given SKU is available or not.",
+          "enum": [
+            "true",
+            "false"
+          ],
+          "x-ms-enum": {
+            "name": "SkuAvailability",
+            "modelAsString": false
+          }
+        },
+        "reason": {
+          "type": "string",
+          "description": "Reaon why the SKU is not available.",
+          "enum": [
+            "Invalid",
+            "NotEligible",
+            "InsufficientQuota"
+          ],
+          "x-ms-enum": {
+            "name": "SkuUnavailableReason",
+            "modelAsString": false
+          }
+        },
+        "message": {
+          "type": "string",
+          "description": "Additional error message."
+        }
+      }
+    },
+    "CognitiveServicesResourceKindSetting": {
+      "description": "Resource kind setting entity.",
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/CognitiveServicesAccountKind"
+        },
+        "allowCreate": {
+          "type": "string",
+          "description": "Whether allow user to create resource.",
+          "enum": [
```

Comment:
the suggestion from AutoRest is to evaluate if the property is really a boolean or not, the intent is to consider if there could be more than 2 values possible for the property in the future or not, if the answer is no, then a boolean is fine, if the answer is yes, there could be other values, then make it an enum with values that represent what those should be. The idea is that it can help avoid breaking changes if you want to add one more possible value, naming the values "true", "false" I believe defeats this purpose.